queries/modules/io/28/modules/xbrl/components.jq

```
@@ -314,31 +324,40 @@ as object*
  : when missing.</p>
  :
  :
- : @param $report-or-id a report or its RID,
+ : @param $component a component object.
+ : @param $hypercube the name of a hypercube.
  : @param $options <a href="facts#standard_options">standard fact retrieving options</a> as well as
  : <ul>
  :   <li>FilterOverride: hypercube dimension specs (as defined as hypercubes:user-defined-hypercube)
  :   to override filters in the report's hypercube.</li>
- :   <li>HypercubeName: picks a different hypercube than the default one.</li>
  : </ul>
  :
+ : @error components:HYPERCUBE-DOES-NOT-EXIST if the specified hypercube is not found.
+ :
  : @return a sequence of facts with populated dimension values.
  :)
 declare function components:facts(
     $component as object,
+    $hypercube as string,
     $options as object?
 )
 as object*
 {
+  if(not $hypercube = (keys($component.Hypercubes), $components:ALL-HYPERCUBES))
+  then error(
+    QName("components:HYPERCUBE-DOES-NOT-EXIST"),
+    "Hypercube " || $hypercube || " was not found."
```

if the hypercube does not exist in the component, an empty sequence cannot be used as string error will be raised, because the hypercube param is empty then: https://github.com/28msec/cellstore/blob/Tables/queries/public/api/facttable-for-component.jq#L123
Should we raise an error there instead?